defs/build.bzl

```diff
@@ -123,25 +121,72 @@ def release_filegroup(name, srcs, visibility = None):
         else:
             basename = src.split("/")[-1]
 
-        md5sum(name = basename + ".md5", src = src, visibility = visibility)
+        srcs_basenames_needing_hashes[src] = basename
         hashes.append(basename + ".md5")
-        sha1sum(name = basename + ".sha1", src = src, visibility = visibility)
         hashes.append(basename + ".sha1")
-        sha512sum(name = basename + ".sha512", src = src, visibility = visibility)
         hashes.append(basename + ".sha512")
+    return hashes
+
+# Creates 3+N rules based on the provided targets:
+# * A filegroup with just the provided targets (named 'name')
+# * A filegroup containing all of the md5, sha1 and sha512 hash files ('name-hashes')
+# * A filegroup containing both of the above ('name-and-hashes')
+# * All of the necessary md5sum, sha1sum and sha512sum rules
+#
+# The targets are specified using the srcs and conditioned_srcs attributes.
+# srcs is expected to be label list.
+# conditioned_srcs is a dictionary mapping conditions to label lists.
+#   It will be passed to select().
+def release_filegroup(name, srcs = None, conditioned_srcs = None, tags = None, visibility = None):
+    if not srcs and not conditioned_srcs:
+        fail("srcs and conditioned_srcs cannot both be empty")
+    srcs = srcs or []
+
+    # A given src may occur in multiple conditioned_srcs, but we want to create the hash
+    # rules only once, so use a dictionary to deduplicate.
+    srcs_basenames_needing_hashes = {}
+
+    hashes = _hashes_for_srcs(srcs, srcs_basenames_needing_hashes)
+    conditioned_hashes = {}
+    if conditioned_srcs:
+        for condition, csrcs in conditioned_srcs.items():
+            conditioned_hashes[condition] = _hashes_for_srcs(csrcs, srcs_basenames_needing_hashes)
+
+    for src, basename in srcs_basenames_needing_hashes.items():
+        md5sum(name = basename + ".md5", src = src, tags = tags, visibility = visibility)
+        sha1sum(name = basename + ".sha1", src = src, tags = tags, visibility = visibility)
+        sha512sum(name = basename + ".sha512", src = src, tags = tags, visibility = visibility)
+
+    if conditioned_srcs:
+        native.filegroup(
```

Comment:
Should we be using output_group instead?

https://docs.bazel.build/versions/master/be/general.html#filegroup.output_group

Reply:
I'm not sure how that helps us?